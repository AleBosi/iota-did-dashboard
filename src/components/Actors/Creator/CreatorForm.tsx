import React, { useState } from "react";
import { Actor } from "../../../models/actor";

interface Props {
  onCreate: (creator: Actor) => void;
}

export default function CreatorForm({ onCreate }: Props) {
  const [name, setName] = useState("");

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    if (!name.trim()) return;
    onCreate({ name } as Actor); // gli altri campi (id, seed, ecc.) gestiti dal parent
    setName("");
  };

  return (
    <form onSubmit={handleSubmit} className="flex gap-2">
      <input
        value={name}
        onChange={e => setName(e.target.value)}
        placeholder="Nome creator"
        className="border px-2 py-1 rounded"
      />
      <button
        type="submit"
        className="bg-blue-500 text-white px-4 py-1 rounded"
        disabled={!name.trim()}
      >
        Aggiungi
      </button>
    </form>
  );
}
