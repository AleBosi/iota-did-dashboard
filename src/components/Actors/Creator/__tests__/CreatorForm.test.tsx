import { render, screen, fireEvent } from "@testing-library/react";
import CreatorForm from "../CreatorForm";

describe("CreatorForm", () => {
  const mockOnCreate = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renderizza campo nome e bottone", () => {
    render(<CreatorForm onCreate={mockOnCreate} />);
    expect(screen.getByPlaceholderText(/nome creator/i)).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /aggiungi/i })).toBeInTheDocument();
  });

  test("il bottone Ã¨ disabilitato se il nome Ã¨ vuoto o solo spazi", () => {
    render(<CreatorForm onCreate={mockOnCreate} />);
    const button = screen.getByRole("button", { name: /aggiungi/i });
    expect(button).toBeDisabled();
    fireEvent.change(screen.getByPlaceholderText(/nome creator/i), { target: { value: "  " } });
    expect(button).toBeDisabled();
  });

  test("il bottone si abilita se viene inserito un nome valido", () => {
    render(<CreatorForm onCreate={mockOnCreate} />);
    const button = screen.getByRole("button", { name: /aggiungi/i });
    fireEvent.change(screen.getByPlaceholderText(/nome creator/i), { target: { value: "Mario" } });
    expect(button).not.toBeDisabled();
  });

  test("chiama onCreate con il nome inserito e resetta il campo", () => {
    render(<CreatorForm onCreate={mockOnCreate} />);
    const input = screen.getByPlaceholderText(/nome creator/i);
    const button = screen.getByRole("button", { name: /aggiungi/i });

    fireEvent.change(input, { target: { value: "Mario" } });
    fireEvent.click(button);

    expect(mockOnCreate).toHaveBeenCalledTimes(1);
    expect(mockOnCreate).toHaveBeenCalledWith({ name: "Mario" });
    expect(input).toHaveValue("");
  });

  test("non chiama onCreate se si prova a inviare con campo vuoto", () => {
    render(<CreatorForm onCreate={mockOnCreate} />);
    fireEvent.click(screen.getByRole("button", { name: /aggiungi/i }));
    expect(mockOnCreate).not.toHaveBeenCalled();
  });

  // Snapshot base
  test("snapshot", () => {
    const { asFragment } = render(<CreatorForm onCreate={mockOnCreate} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
