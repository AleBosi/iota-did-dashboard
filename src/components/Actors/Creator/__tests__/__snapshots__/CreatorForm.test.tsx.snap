// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CreatorForm snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex gap-2"
  >
    <input
      class="border px-2 py-1 rounded"
      placeholder="Nome creator"
      value=""
    />
    <button
      class="bg-blue-500 text-white px-4 py-1 rounded"
      disabled=""
      type="submit"
    >
      Aggiungi
    </button>
  </form>
</DocumentFragment>
`;
